zeebe-io/backport-action
# Add integration-style tests

`npm test` now also runs acceptance tests located in the `test` folder.

## Setup
The acceptance tests are setup using a setup.sh script. Running the test/setup.sh script will create a new directory named `repo` containing a git history that is useful for the acceptance tests.

First run: `(cd test; ./setup.sh)`, and then: `(cd test/repo; git log --graph --oneline)`. The created repo looks like this:
```
*-.   9eb963e (HEAD -> master, release-2) Merge branches 'feature-b' and 'feature-c'
|\ \
| | * 4e2abaa (feature-c) chore: add to c
| | * e5a330b feat: add c
| |/
|/|
* |   b840dd0 (release-1) Merge branch 'feature-a'
|\ \
| * | f416144 (feature-a) feat: add a
|/ /
| * 3d0a6d1 (feature-b) chore: add to b
| * fb01d91 chore: add to b
| * 09d9eed feat: add b
|/
* c23161f init: add README.md
```

You can clean it up again using `(cd test; ./cleanup.sh)`. This cleanup script is also used in the automated acceptance tests.

## Tests
This PR already contains some tests, but these are just an integration tests between the `backport.sh` script and a real git repo. These tests:
- setup a test git repo as described above
- run the backport.sh script against the test repo
- verify that the correct commits are cherry picked to the backport branch
- clean up the test git repo

## Future work
Future work is to expand these tests into full acceptance tests by incorporating the github stuff:
- mock github payload to contain data from test git repo
- run backport.run() instead of backport.sh
- verify that github api is called to create PRs
